### PR TITLE
[data source]: aws_opensearchserverless_collection

### LINKED ISSUE #28463: [New Data Source]: OpenSearch Serverless Collection
### Description

Amazon OpenSearch Serverless is a preview service enabling OpenSearch clusters to be spun up without having to be concerned about the compute or storage requirements.

Collections are used to manage a group of indices that work together to support a given workload.

### Requested Resource(s) and/or Data Source(s)

* `aws_opensearchserverless_collection`

### Potential Terraform Configuration

```terraform
data "aws_opensearchserverless_collection" "example" {
  name        = "example"
}
```

### References

* https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/opensearchserverless@v1.0.2

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by johnsonaj**
### Description

new data source for `aws_opensearchserverless_collection`

### Relations

Closes #28463

### Output from Acceptance Testing

```console
$ make testacc TESTARGS='-run=TestAccOpenSearchServerlessCollectionDataSource_' PKG=opensearchserverless

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/opensearchserverless/... -v -count 1 -parallel 20  -run=TestAccOpenSearchServerlessCollectionDataSource_ -timeout 180m
--- PASS: TestAccOpenSearchServerlessCollectionDataSource_name (707.76s)
--- PASS: TestAccOpenSearchServerlessCollectionDataSource_basic (707.80s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless	711.213s
```

### CODE REVIEW

File: internal/service/opensearchserverless/find.go [hunk 1]
```diff
@@ -60,6 +60,30 @@ func FindCollectionByID(ctx context.Context, conn *opensearchserverless.Client,
...
+				LastRequest: in,
+			}
+		}
+
+		return nil, err
+	}
+
+	if out == nil || out.CollectionDetails == nil || len(out.CollectionDetails) == 0 {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
```
Comment (jar-b) [inline internal/service/opensearchserverless/find.go:82]: If duplicate names are possible with this resource, might be worth a check and return of `tfresource.TooManyResultsError()` here too?

File: website/docs/d/opensearchserverless_collection.html.markdown [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - Amazon Resource Name (ARN) of the collection.
+* `collection_endpoint` - Collection-specific endpoint used to submit index, search, and data upload requests to an OpenSearch Serverless collection.
+* `created_date` - Date the Collection was created.
+* `dashboard_endpont` - Collection-specific endpoint used to access OpenSearch Dashboards.
+* `description` - (Optional) Description of the collection.
```
Comment (jar-b) [inline website/docs/d/opensearchserverless_collection.html.markdown:38]:
```suggestion
* `description` - Description of the collection.
```

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
$ make testacc TESTARGS='-run=TestAccOpenSearchServerlessCollectionDataSource_' PKG=opensearchserverless
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/opensearchserverless/... -v -count 1 -parallel 20  -run=TestAccOpenSearchServerlessCollectionDataSource_ -timeout 180m

--- PASS: TestAccOpenSearchServerlessCollectionDataSource_basic (727.35s)
--- PASS: TestAccOpenSearchServerlessCollectionDataSource_name (737.28s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless       740.608s
```

File: internal/service/opensearchserverless/find.go (hunk 1 above)
Comment (johnsonaj) [reply internal/service/opensearchserverless/find.go:82]: @jar-b good call. Added that check

**Pull request merged**